$OpenBSD: patch-storage_mozStorageConnection_cpp,v 1.5 2020/02/13 07:41:53 landry Exp $

set secure_delete PRAGMA on by default, instead of using internal sqlite copy
https://bugzilla.mozilla.org/show_bug.cgi?id=546162

Brought over from www/firefox-esr

Index: storage/mozStorageConnection.cpp
--- storage/mozStorageConnection.cpp.orig
+++ storage/mozStorageConnection.cpp
@@ -782,6 +782,13 @@ nsresult Connection::initializeInternal() {
     return convertResultCode(srv);
   }
 
+  srv = ::sqlite3_exec(mDBConn, "PRAGMA secure_delete = ON", NULL, NULL, NULL);
+  if (srv != SQLITE_OK) {
+    ::sqlite3_close(mDBConn);
+    mDBConn = nullptr;
+    return convertResultCode(srv);
+  }
+
   // Set the synchronous PRAGMA, according to the preference.
   switch (Service::getSynchronousPref()) {
     case 2:
